### DISCUSSION TITLE
complete support for suspicious users' messages

### DISCUSSION THREAD

**Discussion by CyberDruga** [Chatterino/chatterino2#5506] [Ideas] [answered]
The PR #5056 adds support for showing messages from restricted users and suspicious users updates, but not messages because of some message duplication problems. 

I wanna ask if it is possible to mark messages from suspicious users (not just restricted). I have some Twitch mod friends who would love to get into Chatterino, but are waiting for just this feature to get in.

**Comment by Mm2PL** [comment]
We cannot do that. Twitch does not have an API for this. Twitch might give us one after years of waiting (and it will probably not be usable because it might happen to be restricted to broadcaster-usage only).

**Reply by CyberDruga** [reply]
does that means the `low-trust-users` topic from Twitch Pubsub only shows messages from restricted users and not for suspicious users ? Because what made me think otherwise was the first TODO in #5056, where it sounds like that was not implemented yet because of complications on how it would duplicate the message when using IRC chat.

There's also #4642 that also state they didn't tried implementing sending monitored messages because in their words "it would send 2 messages".

I confess I haven't even tried to implement it in something I know ( I don't know C or C++ ) to see if it is possible, but looking at those 2 PR messages it seems it is.

**Reply by iProdigy** [reply]
I think mm2 meant there is no official api endpoint for the `/monitor` or `/restrict` commands yet

**Reply by CyberDruga** [reply]
ah ok, I might not have been clear with my initial message, sorry about that.

**Comment by iProdigy** [comment, marked as answer]
Flagging monitored messages using the pubsub topic _is_ possible, but involves more complexity than rendering restricted messages currently (especially if we want to accurately render emotes, links, mentions)

This feature is currently on-hold until we add EventSub support (step 1 is #4962), as pubsub is deprecated and the chat-over-eventsub parser could also be used for suspicious messages once Twitch fixes https://github.com/twitchdev/issues/issues/947

**Reply by CyberDruga** [reply]
alright, I understand it now. Thanks both of you for the answers.
